type-check.ts

```diff
@@ -203,34 +218,78 @@ export function tcStmt(env : GlobalTypeEnv, locals : LocalTypeEnv, stmt : Stmt<n
  * @param value Type of the value passed into this destructure
  */
 function tcDestructure(env: GlobalTypeEnv, locals: LocalTypeEnv, destruct: Destructure<null>, value: Type): Destructure<Type> {
-  let values: Type[];
-  
-  if (destruct.isDestructured) {
-    throw new TypeCheckError("Destructured assignment not implemented");
-  } else if (destruct.targets.length === 1) {
-    values = [value];
+
+  if (!destruct.isDestructured) {
+    let target = destruct.targets[0];
+    return {
+      isDestructured: false,
+      targets: [{
+        target: tcAssignable(env, locals, target.target),
+        starred: target.starred,
+        ignore: target.ignore
+      }]
+    };
```

Comment:
These are good points. However, I feel like there's little overlap in logic between type checking a destructured vs nondestructured assignment, and processing them separately makes the code easier to understand and reason about. For example:
- When processing a nondestructured assignment, we look at the overall type of `value`, whereas in the destructured assignment we need to perform additional processing to access the actual types we care about (lines 235 - 241).
- When processing a nondestructured assignment, we only need to call `isAssignable` once for the singular value, so there's no need to incorporate any looping or mapping mechanism.
The only logic they do share in common is ensuring the type(s) of the value(s) and the type(s) of the assign target(s) are compatible, and we've extracted that out into `isAssignable`. Therefore, I think it's better to consider the cases separately.

As to checking ignore, I'm still not convinced we should truly ignore assignments to `_`. If valid Python still tracks the assignments, I feel like we should too.

And yeah, I forgot to call `isAssignable` inside that initial `if` statement. Definitely need to fix that.

Reply:
yeah, that's the exact reason why the non-destruct version did `values = [value]` since then we have a common interface for checking against targets. I agree there are some differences, but I would still say keeping the code DRY is more important in this case, especially given how the current change lost a lot of the type checks (further demonstrating how easy it is to make changes in one spot without reflecting them in a second location).

Whether we handle `_` as truly ignored, we still need to implement the same logic in both locations. Alternatively, we could make this entire issue of `_` moot by just not allowing it/treating it as a normal value. I proposed it as ignore since that's how it's traditionally used in vanilla python.